src/TestSupport/TestLogger.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Namshi\Cuzzle\TestSupport;
+
+use Psr\Log\AbstractLogger;
+
+/**
+ * Used for testing purposes.
+ *
+ * It records all records and gives you access to them for verification.
+ *
+ */
+class TestLogger extends AbstractLogger
+{
+    /**
+     * @var array
+     */
+    public $records = [];
+
+    public $recordsByLevel = [];
+
+    /**
+     * @inheritdoc
+     */
+    public function log($level, $message, array $context = []): void
```

Comment:
Should we type hint the input arguments as well? I assume that level and message will both be a string.

Reply:
This is only typed because of a requirement for compatability with the underlying Abstract class.